src/avram/migrator/create_table_statement.cr

```diff
@@ -71,6 +74,13 @@ class Avram::Migrator::CreateTableStatement
       .build
   end
 
+  macro composite_primary_key(*columns)
+    {% if columns.size < 2 %}
+    {% raise "composite_primary_key expected at least two primary keys, instead got #{columns.size}" %}
+    {% end %}
+    constraints << "  PRIMARY KEY ({{columns.join(", ").id}})"
```

Comment:
Can you explain why you added this `constraints` array instead of adding it to the rows array?

Reply:
It's because constraints must be at the end of create table declaration, so if it was in row, the user should call `composite_primary_key` at the end of the migration.
So user couldn't do a migration like this:
```crystal
def migrate
  create table_for(UserPost) do
    add_belongs_to user : User, on_delete: :cascade
    add_belongs_to post : Post, on_delete: :cascade
    composite_primary_key :user_id, :post_id

    add example : String
  end
end
```
But should do this:
```crystal
def migrate
  create table_for(UserPost) do
    add_belongs_to user : User, on_delete: :cascade
    add_belongs_to post : Post, on_delete: :cascade

    add example : String

    composite_primary_key :user_id, :post_id
  end
end
```
And I find that's sad that we can't put the `composite_primary_key` where we want so I added a constraints array.
But if that's not okay I can change that.

Reply:
@Whaxion I didn't know that it had to be at the end. Good thinking. Could you add a spec showing that a composite key is added at the end of the sql statement even if it is declared first? I'd like to make sure we remember to check for that behavior even if we change things in the future.